assembly/assembly-wsmaster-war/pom.xml

```diff
@@ -1,102 +1,130 @@
+<!-- Copyright (c) 2012-2016 Codenvy, S.A. All rights reserved. This program 
+    and the accompanying materials are made available under the terms of the 
+    Eclipse Public License v1.0 which accompanies this distribution, and is available 
+    at http://www.eclipse.org/legal/epl-v10.html Contributors: Codenvy, S.A. 
+    - initial API and implementation -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <artifactId>almighty-ide-assembly-parent</artifactId>
         <groupId>com.redhat.che</groupId>
-        <version>1.0.0-SNAPSHOT</version>
+        <version>${redhat.che.version}</version>
+        <relativePath>..</relativePath>
     </parent>
     <artifactId>almighty-ide-assembly-wsmaster-war</artifactId>
     <packaging>war</packaging>
     <name>Almighty IDE :: Compiling WS Master WAR</name>
     <properties>
-        <generated.sources.directory>${project.build.directory}/generated-sources/gen</generated.sources.directory>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <original-project-basedir>assembly/${original-project-name}</original-project-basedir>
+        <original-project-name>assembly-wsmaster-war</original-project-name>
     </properties>
     <dependencies>
         <dependency>
             <groupId>aopalliance</groupId>
             <artifactId>aopalliance</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-        </dependency>
-        <dependency>
             <groupId>com.google.inject</groupId>
             <artifactId>guice</artifactId>
         </dependency>
         <dependency>
             <groupId>com.google.inject.extensions</groupId>
-            <artifactId>guice-assistedinject</artifactId>
+            <artifactId>guice-multibindings</artifactId>
         </dependency>
         <dependency>
             <groupId>com.google.inject.extensions</groupId>
-            <artifactId>guice-multibindings</artifactId>
+            <artifactId>guice-persist</artifactId>
         </dependency>
         <dependency>
             <groupId>com.google.inject.extensions</groupId>
             <artifactId>guice-servlet</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.h2database</groupId>
-            <artifactId>h2</artifactId>
+            <groupId>com.redhat.che</groupId>
+            <artifactId>che-keycloak-plugin-server</artifactId>
         </dependency>
         <dependency>
             <groupId>javax.inject</groupId>
             <artifactId>javax.inject</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.eclipse.che</groupId>
```

Comment:
Well, no this dependency isn't enough, because the RH war overriden Java classes depend on these dependencies, and depending on the upstream `war` for overlays doesn't bring transitive dependencies.
We could use a `pom`  dependency on the upstream `war`  project, but this currently makes the dependency-checking unhappy (we already tried with @gorkem  So we kept it like that.